airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -354,3 +307,137 @@ def _get_autogenerated_id(response):
+        reference_image_id=None,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionReferenceImageCreateOperator`
+        """
+        client = self.get_conn()
+        self.log.info('Creating ReferenceImage')
+        parent = ProductSearchClient.product_path(project=project_id, location=location, product=product_id)
+
+        response = client.create_reference_image(
+            parent=parent,
+            reference_image=reference_image,
+            reference_image_id=reference_image_id,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+        if not reference_image_id:
+            # Product set id was generated by the API
+            product_set_id = self._get_autogenerated_id(response)
+            self.log.info('Extracted autogenerated ReferenceImage ID from the response: %s', product_set_id)
+
+        return reference_image_id
+
+    @GoogleCloudBaseHook.catch_http_exception
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def delete_reference_image(
+        self,
+        location,
+        product_id,
+        reference_image_id,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionReferenceImageCreateOperator`
+        """
+        client = self.get_conn()
+        self.log.info('Deleting ReferenceImage')
+        name = ProductSearchClient.reference_image_path(
+            project=project_id, location=location, product=product_id, reference_image=reference_image_id
+        )
+        client.delete_reference_image(name=name, retry=retry, timeout=timeout, metadata=metadata)
+
+    @GoogleCloudBaseHook.catch_http_exception
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def add_product_to_product_set(
+        self,
+        product_set_id,
+        product_id,
+        location=None,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionAddProductToProductSetOperator`
+        """
+        client = self.get_conn()
+
+        product_name = ProductSearchClient.product_path(project_id, location, product_id)
+        product_set_name = ProductSearchClient.product_set_path(project_id, location, product_set_id)
+
+        self.log.info('Add Product[name=%s] to Product Set[name=%s]', product_name, product_set_name)
+
+        client.add_product_to_product_set(
+            name=product_set_name, product=product_name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+        self.log.info('Product added to Product Set')
```

Comment:
For the sake of explicitness we could make this log more specific (with Product and Product Set names), just like the one above.